### PR TITLE
feat: add upload/download icon in upload/download stat + allow hiding header row

### PR DESCRIPTION
**Pull request by aidanaden**
Features added:
- adds upload/download icons to upload/download values
- `hide_headers` config attribute to allow for hiding header row in torrent tab (Name, Completion, Download, Upload, etc)

Purpose is to allow for complete minimalism for pro users (me) that are already familiar with usual torrent statistic values

### CODE REVIEW

File: rm-main/src/ui/tabs/torrents/rustmission_torrent.rs [hunk 1]
```diff
@@ -23,12 +25,13 @@ pub struct RustmissionTorrent {
...
-            self.eta_secs.as_str(),
-            self.download_speed.as_str(),
-            self.upload_speed.as_str(),
+            Cell::new(self.torrent_name.as_str()),
+            Cell::new(""),
+            Cell::new(self.size_when_done.as_str()),
+            Cell::new(self.progress.as_str()),
+            Cell::new(self.eta_secs.as_str()),
+            Cell::new(download_speed_format(&self.download_speed)),
+            Cell::new(upload_speed_format(&self.upload_speed)),
```
Comment (micielski) [inline rm-main/src/ui/tabs/torrents/rustmission_torrent.rs:34]: Any particular reason why these are cells now?

File: rm-main/src/ui/tabs/torrents/rustmission_torrent.rs [hunk 2]
```diff
@@ -49,12 +52,13 @@ impl RustmissionTorrent {
         }
 
         Row::new([
-            torrent_name_line,
-            Line::from(self.size_when_done.as_str()),
-            Line::from(self.progress.as_str()),
-            Line::from(self.eta_secs.as_str()),
-            Line::from(self.download_speed.as_str()),
-            Line::from(self.upload_speed.as_str()),
+            Cell::new(self.torrent_name.as_str()),
```
Comment (micielski) [inline rm-main/src/ui/tabs/torrents/rustmission_torrent.rs:55]: This won't be styled when searching, so use torrent_name_line

File: rm-main/src/ui/tabs/torrents/mod.rs [hunk 3]
```diff
@@ -120,11 +120,17 @@ impl TorrentsTab {
             .general
             .accent_color);
 
-        let table_widget = Table::new(torrent_rows, table_manager_lock.widths)
-            .header(Row::new(
-                table_manager_lock.header().iter().map(|s| s.as_str()),
-            ))
-            .highlight_style(highlight_table_style);
+        let table_widget: Table;
```
Comment (micielski) [inline rm-main/src/ui/tabs/torrents/mod.rs:123]: Ratatui's widgets use lite builder pattern so you can write:
```rust
let table_widget = Table::new(torrent_rows, table_manager_lock.widths)  
                           .highlight_style(highlight_table_style)
```
and then decide whether to add the headers in the ifs, in order to not repeat code

**Review by micielski — CHANGES_REQUESTED** [review]
If hide_header option is in the [general] table in toml, and not in something like [torrents_tab], can you make it apply to search tab also?

File: rm-main/src/ui/tabs/torrents/mod.rs (hunk 3 above)
Comment (aidanaden) [reply rm-main/src/ui/tabs/torrents/mod.rs:123]: Screenshot 2024-06-23 at 6 02 52 PM

not really sure how to deal with this error? the only way i was able to fix it was with this way of doing if-else .. what would u recommend?

Comment (aidanaden) [reply rm-main/src/ui/tabs/torrents/mod.rs:123]: Screenshot 2024-06-23 at 6 05 13 PM

this works but i really didn't want to clone .. is this a better alternative?

File: rm-main/src/ui/tabs/torrents/rustmission_torrent.rs (hunk 1 above)
Comment (aidanaden) [reply rm-main/src/ui/tabs/torrents/rustmission_torrent.rs:34]: ps this was from a previous compiler error, will revert

File: rm-main/src/ui/tabs/torrents/mod.rs (hunk 3 above)
Comment (micielski) [reply rm-main/src/ui/tabs/torrents/mod.rs:123]: This should work:
```rust
let table_widget = {
  let table = Table::new(...).highlight_style(...);
  if hide_headers {
     table.header(...)
  } else {
    table
 }
}
```

**Comment by micielski** [comment]
Could you change the config option name from `hide_headers` to `headers_hide`?

**Review by micielski — CHANGES_REQUESTED** [review]
image
There's a missing B
Fix tthis

**Comment by aidanaden** [comment]
> image There's a missing B Fix tthis

is this the torrent upload speed? or the overall upload speed bottom stat?

**Pull request merged**